sdk/search/Azure.Search.Documents/src/Models/SearchContinuationToken.cs

```diff
@@ -0,0 +1,104 @@
+        {
+            // There's no continuation token if there's no next page
+            if (nextPageUri == null || nextPageOptions == null)
+            {
+                return null;
+            }
+            using MemoryStream buffer = new MemoryStream();
+            using (Utf8JsonWriter writer = new Utf8JsonWriter(buffer))
+            {
+                writer.WriteStartObject();
+                writer.WriteString(s_apiVersionEncodedName, apiVersion.ToVersionString());
+                writer.WriteString(s_nextLinkEncodedName, nextPageUri.ToString());
+                writer.WritePropertyName(s_nextPageParametersEncodedName);
+                writer.WriteObjectValue(nextPageOptions);
+                writer.WriteEndObject();
+            }
+            return Convert.ToBase64String(buffer.ToArray());
+        }
+
+        /// <summary>
+        /// Parses a continuation token and returns the corresponding to the
+        /// next page's <see cref="SearchOptions"/>.
+        /// </summary>
+        /// <param name="continuationToken">
+        /// The serialized continuation token.
+        /// </param>
+        /// <returns>The continuation token's next page options.</returns>
+        public static SearchOptions Deserialize(string continuationToken)
+        {
+            Argument.AssertNotNullOrEmpty(continuationToken, nameof(continuationToken));
+            byte[] decoded = Convert.FromBase64String(continuationToken);
+            try
+            {
+                using JsonDocument json = JsonDocument.Parse(decoded);
+                if (json.RootElement.ValueKind == JsonValueKind.Object &&
+                    json.RootElement.TryGetProperty(ApiVersionName, out JsonElement apiVersion) &&
+                    apiVersion.ValueKind == JsonValueKind.String &&
+                    // Today we only validate against a single known version,
+                    // but in the future we may want to support a range of
+                    // valid continuation token serialization formats.  This
+                    // will need to be updated accordingly.
+                    string.Equals(
+                        apiVersion.GetString(),
+                        SearchClientOptions.ContinuationTokenVersion.ToVersionString(),
+                        StringComparison.OrdinalIgnoreCase) &&
+                    json.RootElement.TryGetProperty(NextPageParametersName, out JsonElement nextPageParams))
+                {
+                    // We only use the nextPageParameters because we do all of
+                    // our searching via HTTP POST requests
+                    return SearchOptions.DeserializeSearchOptions(nextPageParams);
+                }
+            }
+            catch (JsonException)
+            {
+            }
+
+            throw new ArgumentException("Invalid continuation token", nameof(continuationToken));
```

Comment:
Given this whole class is internal, I think it's confusing to throw an `ArgumentException`. Besides, allowing the exception above to throw out is both consistent with all our other deserialization code and provides context about what exactly is wrong. And because the parameter name here is inconsequential to the user anyway, it's certainly *no worse* - only better because it also has more information about what exactly is wrong.

Reply:
The parameter name `continuationToken` is used by all of its (one) caller(s).  It felt cleaner to throw ArgEx here than throw for some conditions/return null and then catch/rewrap in ArgEx at the call site.  We also want to treat these continuationTokens opaquely (for the moment, at least) so I don't want to get specific that certain fields weren't found, etc.  We're going to revisit this in the next preview for consistency across languages.